src/core/Akka.TestKit/TestKitBase_Receive.cs

```diff
@@ -253,20 +253,33 @@ public async ValueTask<object> ReceiveOneAsync(TimeSpan? max = null, Cancellatio
             if (maxDuration.IsZero())
             {
                 ConditionalLog(shouldLog, "Trying to receive message from TestActor queue. Will not wait.");
-                var didTake = _testState.Queue.TryTake(out var item, cancellationToken);
+                var didTake = _testState.Queue.Reader.TryRead(out var item);
                 take = (didTake, item);
             }
             else if (maxDuration.IsPositiveFinite())
             {
                 ConditionalLog(shouldLog, "Trying to receive message from TestActor queue within {0}", maxDuration);
-                take = await _testState.Queue.TryTakeAsync((int)maxDuration.TotalMilliseconds, cancellationToken)
-                    ;
+                var delayTask = Task.Delay(maxDuration, cancellationToken);
```

Comment:
since `maxDuration` is infinite, it is possible for the `Delay` `Task` to survive until the end of the test process, causing a memory leak. For this to be leak safe, we'd need to use a linked `CancellationTokenSource`, use that token, and `Cancel` and `Dispose` of it with a try..catch

Reply:
`maxDuration` cannot be infinite here - that is explicitly checked by `IsPositiveFinite`. The `TimeSpan.Ticks` value is `-1` when it's infinite.